compat/mingw.c

```diff
@@ -475,6 +475,50 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+/* rename and unlink a file. This is needed for the ability to recreate a file
+ * with the same name, for example on checkout which unlinks and recreates the file */
+static int rename_and_unlink(const wchar_t *pathname, int (*unlinker)(const wchar_t *))
+{
+	int res;
+	/* leave some space for the .unlink_XXXXXX extension */
+	wchar_t temppath[MAX_LONG_PATH + 15];
+	*temppath = 0;
+
+	wcsncat(temppath, pathname, sizeof(temppath));
+	wcsncat(temppath, L".unlink_XXXXXX", sizeof(temppath));
+	if (!_wmktemp(temppath) || !MoveFileExW(pathname, temppath, 0))
+		return 1;
+	res = unlinker(temppath);
+	if (res) {
+		const DWORD err = GetLastError();
+		MoveFileExW(temppath, pathname, 0);
+		SetLastError(err);
+	}
+	return res;
+}
+
+static int needs_rename_before_unlink(const wchar_t *pathname, HANDLE *handle)
+{
+	/* If the file is opened by another process, rename it before unlinking */
+	*handle = CreateFileW(pathname, GENERIC_WRITE, FILE_SHARE_DELETE, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
+	return *handle == INVALID_HANDLE_VALUE && is_file_in_use_error(GetLastError());
+}
+
+static int safe_unlink(const wchar_t *wpathname, int (*unlinker)(const wchar_t *))
+{
+	int res;
+	DWORD err;
+	HANDLE handle;
+
+	if (needs_rename_before_unlink(wpathname, &handle))
+		return rename_and_unlink(wpathname, unlinker);
+	res = unlinker(wpathname);
+	err = GetLastError();
+	CloseHandle(hnd);
```

Comment:
`s/hnd/handle/`? The current code does not build.

Reply:
```suggestion
	CloseHandle(handle);
```

Reply:
Done, sorry. I did the change on linux and did not test the build :)

Reply:
No worries, that's why we have automated build & test ;-) Thank you for fixing it!

I'll wait for @drizzd's feedback, and I also offered a couple more reviewer comments myself.